### ISSUE TITLE
Installing types-requests breaks urllib3 2.0 type hints

### ISSUE THREAD

**Issue by pquentin** [python/typeshed#10786]
requests 2.31.0 depends on urllib3, and [supports both urllib3 1.21.x and urllib3 2.0.x](https://github.com/psf/requests/blob/ee93fac6b2f715151f1aa9a1a06ddba9f7dcc59a/setup.py#L64). requests does not have inline type hints. urllib3 1.x did not have inline type hints, we introduced them in urllib3 2.0, as correctly identified here:

https://github.com/python/typeshed/blob/4dad5311d56e0b90ab82756a8c2a9ce645a788b1/stubs/urllib3/METADATA.toml#L3

However, when taking advantages of urllib3 2.0 type hints in a file like this:

```python3
import urllib3.connection


class HTTPSConnection(urllib3.connection.HTTPSConnection):
    def connect(self) -> None:
        super().connect()
        self.foo = "bar"
```

`mypy --strict t.py` reports no error, *unless* I run `pip install types-requests`, which installs types-requests 2.31.0.6 AND types-urllib3 1.26.25.14. At this point, the typeshed package takes precedence even if I'm using urllib3 2.0.5 at runtime, and `mypy --strict t.py` reports:

```
t.py:6: error: Call to untyped function "connect" in typed context  [no-untyped-call]
Found 1 error in 1 file (checked 1 source file)
```

The manual fix is to manually uninstall types-urllib3, but is there a way to make this work without manual intervention?

**Comment by AlexWaygood** [comment]
We'd like to change our `types-requests` package so that it depends on `urllib3` rather than `types-urllib3`. Unfortunately, doing so breaks our tests due to a limitation of how our CI is set up -- our tests start complaining about differences between our outdated stubs for `urllib3` and the inline types in the runtime package for `urllib3`, and it's all a bit of a mess.

However, good news! In four days, we're due to remove our stubs for `urllib3` from typeshed (#6898). Once we do so, we can easily change `types-requests` so that it depends on `urllib3` rather than `types-urllib3`, which will resolve this issue for you.

TL;DR: closing as a duplicate of #10142 ;)

**Comment by AlexWaygood** [comment]
@pquentin hopefully this is resolved with  `types-requests==2.31.0.7`?
